cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java

```diff
@@ -167,11 +184,14 @@ public LoadMonitorState state() {
     // Get valid snapshot window number and populate the monitored partition map.
     // We do this primarily because the checker and aggregator are not always synchronized.
     int numValidSnapshotWindows = 0;
-    SortedMap<Long, Double> loadSnapshotsWindows = new TreeMap<>();
+    boolean hasInvalidSnapshotWindows = false;
+    SortedMap<Long, Double> loadSnapshotsWindows = new TreeMap<>((w1, w2) -> Long.compare(w2, w1));
```

Comment:
Why is the ordering of the snapshots being changed?

Reply:
You might want to use Comparator.reverseOrder() instead.

Reply:
We changed the window order in MetricSampleAggregator to simplify the logic. So it seems more consistent to keep the same order here. Good suggestion about `Comparator.reverseOrder()`.